### ISSUE TITLE
[Bug]: FileReadObservation.init() got an unexpected keyword argument 'impl_source'

### ISSUE THREAD

**Issue by avi12** [All-Hands-AI/OpenHands#5741]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Describe the bug and reproduction steps

https://www.all-hands.dev/share?share_id=89cc5c6565e8372271be4bb803ec6eff0cee8201800be7514a63500edda88490

### OpenHands Installation

Docker command in README

### OpenHands Version

0.16.1

### Operating System

WSL on Windows

**Comment by malhotra5** [comment]
+1 on this ([Workflow log](https://github.com/malhotra5/test-repo/actions/runs/12450819876/job/34758108971))

![image](https://github.com/user-attachments/assets/3e55de0a-c4bc-465d-8920-3894253e7930)

**Comment by avi12** [comment]
It seems like this particular issue was solved with the bot's update

**Comment by malhotra5** [comment]
Reopening this issue, still facing the issue on `main`

Sample [workflow log](https://github.com/malhotra5/test-repo/actions/runs/12455206766/job/34767337481)

**Comment by openhands-agent** [comment]
[OpenHands](https://github.com/All-Hands-AI/OpenHands) started fixing the issue! You can monitor the progress [here](https://github.com/All-Hands-AI/OpenHands/actions/runs/12455263809).

**Comment by openhands-agent** [comment]
An attempt was made to automatically fix this issue, but it was unsuccessful. A branch named 'openhands-fix-issue-5741' has been created with the attempted changes. You can view the branch [here](https://github.com/All-Hands-AI/OpenHands/tree/openhands-fix-issue-5741). Manual intervention may be required.

**Comment by enyst** [comment]
To note, this looks like a version conflict:
- this was merged after 0.16.1 release https://github.com/All-Hands-AI/OpenHands/pull/5508
  - it does indeed introduce two new attributes, which don't exist in the 0.16.1 version of the event
- the workflow uses the 0.16.1 version
<img width="806" alt="image" src="https://github.com/user-attachments/assets/45031fd8-78dd-4c5d-913e-9b2f4da22c48" />

**Comment by malhotra5** [comment]
Ah gotcha, how about this [workflow log](https://github.com/malhotra5/test-repo/actions/runs/12455206766/job/34767337481)? It's using `main`

**Comment by enyst** [comment]
It looks like [this](https://github.com/malhotra5/test-repo/actions/runs/12455206766/job/34767337481#step:10:140) is not main, it's 0.16.1... 🤔

**Comment by enyst** [comment]
I mean line 139 there

**Comment by malhotra5** [comment]
Ah I see the confusion - the previous step in the workflow checks for the latest version

![image](https://github.com/user-attachments/assets/363ab4b2-7437-417a-a845-e67d513db22e)

However for the step that installs the package, I forced it to install from main 
![image](https://github.com/user-attachments/assets/c6b86e78-076f-4b9b-9984-54fb9b095612)


When I use the released version 0.16.1, this issue doesn't occur ([another workflow log
](https://github.com/malhotra5/test-repo/actions/runs/12455429444/job/34767815706))

It seems like this is an issue only on `main`

**Comment by enyst** [comment]
Hmm, how about the sandbox? This code above is running in the sandbox 🤔

**Comment by malhotra5** [comment]
I believe the resolver runs on `eventstream` instead of `sandbox` runtime; am I interpreting this correctly?
![image](https://github.com/user-attachments/assets/afc3aa4f-05b8-4501-aa5d-5ea96b2dcfa2)

**Comment by enyst** [comment]
The "eventstream runtime" is a docker sandbox, which communicates using an API sending/receiving events from the sandbox. It's not local, it's in docker.

> EDIT: we can confirm this by running fix-me vs fix-me-experimental on an issue; I believe fix-me-experimental will fail

I ran fix-me-experimental [here](https://github.com/All-Hands-AI/OpenHands/pull/5740) last night, it worked. The difference might be that your test repo is an external repo, while the successful workflow worked on this repo?

**Comment by openhands-agent** [comment]
[OpenHands](https://github.com/All-Hands-AI/OpenHands) started fixing the issue! You can monitor the progress [here](https://github.com/All-Hands-AI/OpenHands/actions/runs/12455745077).

**Comment by malhotra5** [comment]
Possibly! Did a quick test, was able to reproduce the error again on this repo; there may be a PR that caused a regression in the last day or so

**Comment by enyst** [comment]
Oh, it failed with the same error!

I saw this: https://github.com/All-Hands-AI/OpenHands/actions/runs/12455745077/job/34768492590#step:10:37

**Comment by ryanhoangt** [comment]
I think the issue is due to the resolver used a specified runtime image, which skips the runtime image build process:

https://github.com/All-Hands-AI/OpenHands/blob/4a8bf3d2d0b3a032b16962b62f2cce65c4fe1850/openhands/resolver/resolve_issue.py#L593-L596

One way to address might be to not specify this runtime image at all, similar to the dev workflow, which can add some build time overhead to the resolver, but can keep it up to date with all the changes in `main`. What do you guys think?

**Comment by gordicaleksa** [comment]
+1

I'll revert to the last known stable commit as i'm just getting familiar with the project :)

**Comment by enyst** [comment]
> I think the issue is due to the resolver using a specified runtime image, which skipped the runtime image build process:
> 
> https://github.com/All-Hands-AI/OpenHands/blob/4a8bf3d2d0b3a032b16962b62f2cce65c4fe1850/openhands/resolver/resolve_issue.py#L593-L596
> 
> One way to address might be to not specify this runtime image at all, similar to the dev workflow, which can add some build time overhead to the resolver, but can keep it up to date with all the changes in `main`. What do you guys think?

Aha, that was the code. I agree! I think we should fix it as you suggest: let it build the runtime it needs.

Because we now use `main` in our fairly regularly used workflow, and also, importantly, other repos also build on top of our `main` workflow. Conflicts like this between the sandbox image and main can be pretty bad and they're almost blocking for external workflows.

Please note that this isn't always: this is for `fix-me-experimental` and equivalent macro (at openhands-agent-exp), which run on `main`. For the rest, the regular `fix-me` workflow (and equivalent macro), we can keep the released runtime, so there are no build times concerns there. Those run on the `pypi` version.

**Comment by enyst** [comment]
@glgh @gordicaleksa If you encounter this without running the resolver, can you please remove all openhands-related images from docker, and let it re-download?

**Comment by ryanhoangt** [comment]
@enyst Yeah, let me have a closer look and try a PR for that! This is a good opportunity for me to learn more about the resolver! 😅
